tests/api/iotrace_parser.py

```diff
@@ -0,0 +1,173 @@
+        ret += f"duration: {self.read}"
+        ret += f"duration: {self.write}"
+        ret += f"duration: {self.discard}"
+        ret += f"duration: {self.flush}"
+        ret += f"duration: {self.total}"
+
+        return ret
+
+
+class DeviceStats:
+    def __init__(self, trace):
+        self.id = trace["id"]
+        self.name = trace["name"]
+        self.size = Size(float(trace["size"]), Unit.Blocks512)
+
+    def __str__(self):
+        ret = f"device: id: {self.id}, name: {self.name}, size {self.size}\n"
+        return ret
+
+
+class IOStatistics:
+    def __init__(self, trace):
+        self.size = SectorStat(trace["size"])
+        self.latency = TimeStat(trace["latency"])
+        self.metrics = Metrics(trace["metrics"])
+        self.errors = int(trace["errors"])
+
+    def __str__(self):
+        ret = f"{self.size}"
+        ret += f"{self.latency}"
+        ret += f"{self.metrics}"
+        ret += f"errors {self.errors}"
+
+        return ret
+
+
+class TimeStat:
+    def __init__(self, trace):
+        # TODO class with nanoseconds needed
+        self.average = int(trace["average"])
+        self.min = int(trace["min"])
+        self.max = int(trace["max"])
+        self.count = int(trace["count"])
+        self.total = int(trace["total"])
+        self.percentiles = Percentiles(
+            trace["percentiles"], "of reqs have latency lower than"
+        )
+
+    def __str__(self):
+        ret = f"average latency: {self.average} ns\n"
+        ret += f"min latency: {self.min} ns\n"
+        ret += f"max latency: {self.max} ns\n"
+        ret += f"count: {self.count}\n"
+        ret += f"total: {self.min} ns\n"
+        ret += f"{self.percentiles}"
+
+        return ret
+
+
+class SectorStat:
+    def __init__(self, trace):
+        self.average = Size(float(trace["average"]), Unit.Blocks512)
+        self.min = Size(float(trace["min"]), Unit.Blocks512)
+        self.max = Size(float(trace["max"]), Unit.Blocks512)
+        self.count = Size(float(trace["count"]), Unit.Blocks512)
+        self.total = Size(float(trace["total"]), Unit.Blocks512)
+        self.percentiles = Percentiles(
+            trace["percentiles"], "of reqs touches no more sectors than"
+        )
+
+    def __str__(self):
+        ret = f"average size: {self.average} sector(s)\n"
+        ret += f"min size: {self.min} sector(s)\n"
+        ret += f"max size: {self.max} sector(s)\n"
+        ret += f"count: {self.count}\n"
+        ret += f"total: {self.min} sector(s)\n"
+        ret += f"{self.percentiles}"
+
+        return ret
+
+
+class ServiceStats:
+    def __init__(self, trace):
+        self.metrics = Metrics(trace["metrics"])
+        self.errors = int(trace["errors"])
+
+    def __str__(self):
+        ret = f"metrics {self.metrics}"
+        ret += f"errors {self.errors}\n"
+
+        return ret
+
+
+class Percentiles:
+    def __init__(self, trace, message="of population have value lower than"):
+        TestRun.LOGGER.info(f"{trace}")
```

Comment:
Why are we printing here?

Reply:
It was printed for debug purposes. I removed it in latest version.